cmd/collector/app/span_processor_test.go

```diff
@@ -807,3 +815,105 @@ func TestSpanProcessorWithOnDroppedSpanOption(t *testing.T) {
 	require.EqualError(t, err, processor.ErrBusy.Error())
 	assert.Equal(t, []string{"op3"}, droppedOperations)
 }
+
+func optionsWithPorts(portHttp string, portGrpc string) *cFlags.CollectorOptions {
+	opts := &cFlags.CollectorOptions{
+		OTLP: struct {
+			Enabled bool
+			GRPC    configgrpc.ServerConfig
+			HTTP    confighttp.ServerConfig
+		}{
+			Enabled: true,
+			HTTP: confighttp.ServerConfig{
+				Endpoint: portHttp,
+			},
+			GRPC: configgrpc.ServerConfig{
+				NetAddr: confignet.AddrConfig{
+					Endpoint:  portGrpc,
+					Transport: confignet.TransportTypeTCP,
+				},
+			},
+		},
+	}
+	return opts
+}
+
+func TestOTLPReceiverWithV2Storage(t *testing.T) {
+	// Setup mock writer and expectations
+	mockWriter := mocks.NewWriter(t)
+	traces := ptrace.NewTraces()
+	rSpans := traces.ResourceSpans().AppendEmpty()
+	sSpans := rSpans.ScopeSpans().AppendEmpty()
+	span := sSpans.Spans().AppendEmpty()
+	span.SetName("test-trace")
+	byteId := [8]byte{1, 2, 3, 4, 5, 6, 7}
+	span.SetSpanID(byteId)
+
+	mockWriter.On("WriteTraces", mock.Anything, mock.Anything).Run(func(args mock.Arguments) {
+		td := args.Get(1).(ptrace.Traces)
+
+		equal := td.ResourceSpans().Len() == traces.ResourceSpans().Len()
+
+		if equal {
+			rSpan := td.ResourceSpans().At(0)
+			equal = rSpan.ScopeSpans().Len() == 1 &&
+				rSpan.ScopeSpans().At(0).Spans().Len() == 1 &&
+				rSpan.ScopeSpans().At(0).Spans().At(0).Name() == "test-trace" &&
+				rSpan.ScopeSpans().At(0).Spans().At(0).SpanID() == byteId
+		}
+		if !equal {
+			t.Error("Traces don't match expected structure")
+		}
+	}).Return(nil)
+
+	spanProcessor, err := NewSpanProcessor(
+		mockWriter,
+		nil,
+		Options.NumWorkers(1),
+		Options.QueueSize(1),
+		Options.ReportBusy(true),
+	)
+	require.NoError(t, err)
+	defer spanProcessor.Close()
+	logger, _ := testutils.NewLogger()
```

Comment:
if you don't need to inspect logger results (you're ignoring the `buf` return value), use `zaptest.NewLogger(t)` or `zap.NewNop()` instead